sql/driver/driver_test.go

```diff
@@ -202,6 +204,50 @@ func TestPlaceholders(t *testing.T) {
 	}
 }
 
+func TestConnectionSettings(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	s := server.StartTestServer(nil)
+	url := "https://root@" + s.ServingAddr() + "?certs=test_certs"
+	// Create a new Database called t.
+	db, err := sql.Open("cockroach", url)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if _, err := db.Exec(`CREATE DATABASE t`); err != nil {
+		t.Fatal(err)
+	}
+	_ = db.Close()
+
+	// Open a new db client against the server with
+	// a fresh set of connections with the new settings.
+	// We create a database 't' above because you cannot
+	// use sql: "SET DATABASE = t" without the database
+	// existing.
+	url += "&database=t"
+	if db, err = sql.Open("cockroach", url); err != nil {
+		t.Fatal(err)
+	}
+	defer cleanup(s, db)
+	var wq sync.WaitGroup
+	numRoutines := 10
+	wq.Add(numRoutines)
```

Comment:
Let me change this to use a txn after I've merged the multi-statement sql PR. I'll keep this PR around until then.